Core/FieldType/EnhancedSelection/SearchField.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Netgen\Bundle\EnhancedSelectionBundle\Core\FieldType\EnhancedSelection;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition;
+use eZ\Publish\SPI\FieldType\Indexable;
+use eZ\Publish\SPI\Search;
+class SearchField implements Indexable
+{
+    /**
+     * Get index data for field for search backend.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Field $field
+     * @param \eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition $fieldDefinition
+     *
+     * @return \eZ\Publish\SPI\Search\Field[]
+     */
+    public function getIndexData(Field $field, FieldDefinition $fieldDefinition)
+    {
+        $selectionKeys = (array)$field->value->externalData;
+        $selectionIds = [];
+        $selectionNames = [];
+
+        foreach ($fieldDefinition->fieldTypeConstraints->fieldSettings['options'] as $option) {
+            if (in_array($option['identifier'], $selectionKeys)) {
+                $selectionIds[] = $option['id'];
+                $selectionNames[] = $option['name'];
+            }
+        }
+        return [
+            new Search\Field(
+                'enhanced_selection_identifiers',
```

Comment:
Is it required to prefix the fields with `enhanced_selection`? Field names will be generated from field identifier as it is.